packages/plugin-react/src/index.ts

```diff
@@ -167,145 +158,117 @@ export default function viteReact(opts: Options = {}): PluginOption[] {
       }
     },
     async transform(code, id, options) {
+      if (id.includes('/node_modules/')) return
+
+      const [filepath] = id.split('?')
+      if (!filter(filepath)) return
+
       const ssr = options?.ssr === true
-      // File extension could be mocked/overridden in querystring.
-      const [filepath, querystring = ''] = id.split('?')
-      const [extension = ''] =
-        querystring.match(fileExtensionRE) ||
-        filepath.match(fileExtensionRE) ||
-        []
-
-      if (/\.(?:mjs|[tj]sx?)$/.test(extension)) {
-        const isJSX = extension.endsWith('x')
-        const isNodeModules = id.includes('/node_modules/')
-        const isProjectFile =
-          !isNodeModules && (id[0] === '\0' || id.startsWith(projectRoot + '/'))
-
-        const babelOptions = (() => {
-          if (staticBabelOptions) return staticBabelOptions
-          const newBabelOptions = createBabelOptions(
-            typeof opts.babel === 'function'
-              ? opts.babel(id, { ssr })
-              : opts.babel,
+      const babelOptions = (() => {
+        if (staticBabelOptions) return staticBabelOptions
+        const newBabelOptions = createBabelOptions(
+          typeof opts.babel === 'function'
+            ? opts.babel(id, { ssr })
+            : opts.babel,
+        )
+        runPluginOverrides?.(newBabelOptions, { id, ssr })
+        return newBabelOptions
+      })()
+      const plugins = [...babelOptions.plugins]
+
+      const useFastRefresh = !skipFastRefresh && !ssr
+      if (useFastRefresh) {
+        plugins.push([
+          await loadPlugin('react-refresh/babel'),
+          { skipEnvCheck: true },
+        ])
```

Comment:
This line seems to run every file. Before this PR, this part was inside `if ((isJSX || importReactRE.test(code)) && filter(id))`.
I guess this change is for https://github.com/vitejs/vite-plugin-react/issues/83.

Previously we were able to skip babel completely for a non-react file. I'm not sure how much this will impact the performance but I guess this may make a normal react project slower.

Reply:
Oh yes that's a good point.

Maybe we should go one step further and default to only applying the plugin is JSX/TSX file people can use the include to add more to it.